src/rust/engine/src/lib.rs

```diff
@@ -315,11 +315,34 @@ pub extern "C" fn scheduler_metrics(
 ) -> Handle {
   with_scheduler(scheduler_ptr, |scheduler| {
     with_session(session_ptr, |session| {
-      let values = scheduler
+      let mut values = scheduler
         .metrics(session)
         .into_iter()
         .flat_map(|(metric, value)| vec![externs::store_utf8(metric), externs::store_i64(value)])
         .collect::<Vec<_>>();
+      let values_ref = &mut values;
+      if session.should_record_zipkin_spans() {
+        let workunits = &session
+          .get_workunits()
+          .lock()
+          .iter()
+          .map(|workunit| {
+            let workunit_zipkin_trace_info = vec![
+              externs::store_utf8("name"),
+              externs::store_utf8(&workunit.name),
+              externs::store_utf8("start_timestamp"),
+              externs::store_f64(workunit.start_timestamp),
+              externs::store_utf8("end_timestamp"),
+              externs::store_f64(workunit.end_timestamp),
+              externs::store_utf8("span_id"),
+              externs::store_utf8(&workunit.span_id),
```

Comment:
I've made #7371 for a related thought about how to make creating python objects a little more declarative in general, but this is great as is.